src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.TimeDate/Components/TimeAndDateHelper.cs

```diff
@@ -0,0 +1,110 @@
+        /// <param name="dateLong">Show time with seconds (long format)</param>
+        /// <returns>String that identifies the time/date format (<see href="https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring"/>)</returns>
+        internal static string GetStringFormat(FormatStringType targetFormat, bool timeLong, bool dateLong)
+        {
+            switch (targetFormat)
+            {
+                case FormatStringType.Time:
+                    return timeLong ? "T" : "t";
+                case FormatStringType.Date:
+                    return dateLong ? "D" : "d";
+                case FormatStringType.DateTime:
+                    if (timeLong & dateLong)
+                    {
+                        return "F"; // Friday, October 31, 2008 5:04:32 PM
+                    }
+                    else if (timeLong & !dateLong)
+                    {
+                        return "G"; // 10/31/2008 5:04:32 PM
+                    }
+                    else if (!timeLong & dateLong)
+                    {
+                        return "f"; // Friday, October 31, 2008 5:04 PM
+                    }
+                    else
+                    {
+                        // (!timeLong & !dateLong)
+                        return "g"; // 10/31/2008 5:04 PM
+                    }
+
+                default:
+                    return string.Empty; // Windows default based on current culture settings
+            }
+        }
+
+        /// <summary>
+        /// Returns the number week in the month (Used code from 'David Morton' from <see href="https://social.msdn.microsoft.com/Forums/vstudio/en-US/bf504bba-85cb-492d-a8f7-4ccabdf882cb/get-week-number-for-month"/>)
+        /// </summary>
+        /// <param name="date">date</param>
+        /// <returns>Number of week in the month</returns>
+        internal static int GetWeekOfMonth(DateTime date)
+        {
+            DateTime beginningOfMonth = new DateTime(date.Year, date.Month, 1);
+
+            while (date.Date.AddDays(1).DayOfWeek != CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek)
+            {
+                date = date.AddDays(1);
+            }
+
+            return (int)Math.Truncate((double)date.Subtract(beginningOfMonth).TotalDays / 7f) + 1;
+        }
+
+        /// <summary>
+        /// Convert input string to a <see cref="DateTime"/> object in local time
+        /// </summary>
+        /// <param name="input">String with date/time</param>
+        /// <param name="timestamp">The new <see cref="DateTime"/> object</param>
+        /// <returns>True on success, otherwise false</returns>
+        internal static bool ParseStringAsDateTime(in string input, out DateTime timestamp)
+        {
+            if (DateTime.TryParse(input, out timestamp))
+            {
+                // Known date/time format
+                return true;
+            }
+            else if (int.TryParse(input, out int secondsInt))
+            {
```

Comment:
switch back to letter indication as int32 will fail after `03:14:07 UTC 2038-01-19`.

@jaimecbernardo 
Which chars would you use? Thought on `uXXXXX` for Unix time and `fXXXXX` for file time or should we only use a letter for `Windows File Time` number?

Reply:
@franky920920